src/transaction.rs

```diff
@@ -408,6 +449,25 @@ impl Transaction {
 
         self.serialized_size() - witness_data // Witness data size
     }
+
+    pub(crate) fn _outputs_mut(&mut self) -> &mut [Output] {
+        match self {
+            Self::Script { outputs, .. } => outputs.as_mut_slice(),
+            Self::Create { outputs, .. } => outputs.as_mut_slice(),
+        }
+    }
+
+    /// Prepare the transaction for VM initialization
+    ///
+    /// # Steps
+    ///
+    /// 1) Zero all recipient and amount from message outputs
```

Comment:
Should this also zero out variable outputs?

Reply:
https://github.com/FuelLabs/fuel-tx/pull/149/commits/38ebb1a7d60015f6d2c6cda669eb71eb2a53783d